doc/source/advancedexamples.rst

```diff
@@ -1164,8 +1164,12 @@ exponentials of the Trotter decomposition and can be executed on any state.
 Note that in the transverse field Ising model (TFIM) that was used in this
 example is among the pre-coded Hamiltonians in Qibo and could be created as
 a :class:`qibo.core.hamiltonians.SymbolicHamiltonian` simply using the
-``dense=False`` flag. Defining custom Hamiltonians from terms can be more
-complicated, however Qibo simplifies this process by providing the option
+``dense=False`` flag. For more information on the difference between dense
+and non-dense Hamiltonians we refer to the :ref:`Hamiltonians <Hamiltonians>` 
+section.
```

Comment:
Looks good, however I still believe someone will ask "what happens if dense if True and I do trotterization"? Maybe include a sentence about this here.

Reply:
Oh I see. The answer to this is that if `dense=True` then no trotterization will be used because the full Hamiltonian matrix is available so Qibo will directly exponentiate this in order to perform evolution. I added an additional sentence to this paragraph explaining this.

Do you think we should allow trotterization using dense (full matrix) Hamiltonians? Generally the idea is that one can construct the SymbolicHamiltonian which has a more compact term-by-term representation and use this for memory efficient expectation values and Trotter evolution and in case the full matrix is needed it is still possible to do `SymbolicHamiltonian.dense` to get the dense for or `SymbolicHamiltonian.matrix` to get the 2^n x 2^n matrix directly.

Reply:
Indeed, I think the current logics is fine.